lib/index.js

```diff
@@ -12,34 +26,307 @@ class PluginPassportOAuth {
+   *
+   * @param request
+   * @param credentials
+   * @param userId user id in Kuzzle
+   * @param provider
+   * @returns {Promise|Promise.<TResult>}
+   */
+  create (request, credentials, userId, provider) {
+    let user = {};
+
+    return this.exists(request, userId, provider)
+      .then(exists => {
+        if (exists) {
+          return Promise.reject(new this.context.errors.BadRequestError('A strategy already exists for this user.'));
+        }
+        Object.keys(credentials).forEach(attr => {
+          user[attr] = credentials[attr];
+        });
+        user.userId = userId;
+
+        return this.getProviderRepository(provider).create(user);
+      });
+  }
+
+  /**
+   * Update a user in the repository of the corresponding strategy
+   *
+   * @param request
+   * @param credentials
+   * @param userId
+   * @param provider
+   * @returns {Promise|Promise.<TResult>}
+   */
+  update (request, credentials, userId, provider) {
+    return this.getCredentialsFromUserId(userId, provider)
+      .then(document => {
+        if (document === null) {
+          return Promise.reject(new this.context.errors.BadRequestError('A strategy does not exist for this user.'));
+        }
+
+        return this.getProviderRepository(provider).update(
+          Object.assign({_id: document._id}, credentials)
+        );
+      });
+  }
+
+  /**
+   * Delete a user in the repository of the corresponding strategy
+   *
+   * @param request
+   * @param userId
+   * @param provider
+   * @returns {Promise|Promise.<TResult>}
+   */
+  delete (request, userId, provider) {
+    return this.getCredentialsFromUserId(userId, provider)
+      .then(document => {
+        if (document === null) {
+          return Promise.reject(new this.context.errors.BadRequestError('A strategy does not exist for this user.'));
+        }
+
+        return this.getProviderRepository(provider).delete(document._id, {refresh: 'wait_for'});
+      });
+  }
+
+  /**
+   * Check if a userId already exists
+   *
+   * @param request
+   * @param credentials
+   * @param userId
+   * @param provider
+   * @returns {Promise|Promise.<TResult>}
+   */
+  validate (request, credentials, userId, provider) {
+    return this.getProviderRepository(provider).get(credentials._id)
+      .then(result => {
+        if (result === null) {
+          return Promise.resolve(true);
+        }
+
+        if (userId !== result.userId) {
+          return Promise.reject(new this.context.errors.BadRequestError(`Login '${credentials._id}' is already used.`));
+        }
+        return Promise.resolve(true);
+      });
+  }
+
+  /**
+   * Get the credentials of the user.
+   * This do not return the _id of the user
+   *
+   * @param request
+   * @param userId
+   * @param provider
+   * @returns {Promise|Promise.<TResult>}
+   */
+  getInfo (request, userId, provider) {
+    return this.getCredentialsFromUserId(userId, provider)
```

Comment:
`userId` here is the user id in Kuzzle, so we should just call `this.getProviderRepository(provider).get(userId)`